ml_pipeline_engine/dag/manager.py

```diff
@@ -154,128 +246,124 @@ def _filter(u: str, v: str) -> bool:
-
-            # Если в узел попадают параметры, которые являются ошибками, то исполнять этот узел нельзя.
-            # Как следствие, нам нужно завершить исполнение текущей ноды
-            for key_value in input_kwargs.values():
-                if isinstance(key_value, Exception):
-                    raise key_value
+            logger.info('Preparing node for the execution, node_id=%s', node_id)
 
-            result = await self.execute(
+            result = await self.__execute_node(
                 node_id=node_id,
                 force_default=force_default,
-                **input_kwargs,
+                **self._get_node_kwargs(node_id),
             )
 
-            if isinstance(result, Exception):
-                await self.ctx.emit_on_node_complete(node_id=node_id, error=result)
-            else:
-                await self.ctx.emit_on_node_complete(node_id=node_id, error=None)
+            await self.ctx.emit_on_node_complete(node_id=node_id, error=None)
 
-            logger.info('Завершение исполнения ноды, node_id=%s', node_id)
+            logger.info('Getting the result after the execution, node_id=%s', node_id)
             return result
 
         except Exception as ex:
             await self.ctx.emit_on_node_complete(node_id=node_id, error=ex)
-            logger.error('Ошибка исполнения ноды node_id=%s, err=%s', node_id, ex)
+            logger.error('Execution error node_id=%s', node_id, exc_info=ex)
 
-            if is_node_from_success_pool:
-                return NodeErrorType.succession_node_error
+            if is_under_oneof:
+                return ex
 
             raise ex
 
-    async def execute(
+    async def __execute_node(
         self,
         node_id: NodeId,
         force_default: bool,
         **kwargs: t.Any,
     ) -> t.Union[NodeResultT, t.Any]:
         """
-        Запуск ноды с дополнительной обработкой
+        Execute the node by node id with retry protocol
 
         Args:
-            node_id: Идентификатор ноды
-            force_default: Отдать дефолтное значение
-            **kwargs: Ключевые аргументы целевой функции
+            node_id: Node id
+            force_default: Use default result or raise an exception if there isn't default method
+            **kwargs: Key value args for the node
         """
 
         node = self.dag.node_map[node_id]
 
-        retry_policy = DagRetryPolicy(node=node)
+        retry_policy = NodeRetryPolicy(node=node)
 
         n_attempts = 1
         while True:
             try:
-                logger.debug('Начало исполнения узла node_id=%s', node_id)
+                logger.debug('Start execution node_id=%s', node_id)
+
                 if force_default:
-                    return run_node_default(node, **kwargs)
+                    if node.use_default:
+                        return run_node_default(node, **kwargs)
+
+                    raise DefaultMethodDoesNotExistError
```

Comment:
I have a feeling that we should keep track of such things at the beginning of graf creation, I've already had thoughts about the need for pre validation

Reply:
Good point. I changed the behaviour